composer/datasets/cifar10.py

```diff
@@ -1,60 +1,80 @@
 # Copyright 2021 MosaicML. All Rights Reserved.
 
+import textwrap
 from dataclasses import dataclass
+from typing import Optional
 
+import torch.utils.data
 import yahp as hp
 from torchvision import transforms
 from torchvision.datasets import CIFAR10
 
 from composer.core.types import DataLoader
 from composer.datasets.dataloader import DataloaderHparams
-from composer.datasets.hparams import DatasetHparams
+from composer.datasets.hparams import (DatadirHparamsMixin, DatasetHparams, DropLastHparamsMixin, IsTrainHparamsMixin,
+                                       NumTotalBatchesHparamsMixin, ShuffleHparamsMixin, SyntheticBatchesHparamsMixin)
+from composer.datasets.synthetic import SyntheticBatchPairDatasetHparams
 from composer.utils import ddp
 
 
 @dataclass
-class CIFAR10DatasetHparams(DatasetHparams):
+class CIFAR10DatasetHparams(DatasetHparams, ShuffleHparamsMixin, DropLastHparamsMixin, DatadirHparamsMixin,
+                            NumTotalBatchesHparamsMixin, SyntheticBatchesHparamsMixin, IsTrainHparamsMixin):
     """Defines an instance of the CIFAR-10 dataset for image classification.
     
     Parameters:
-        is_train (bool): Whether to load the training or validation dataset.
-        datadir (str): Data directory to use.
         download (bool): Whether to download the dataset, if needed.
-        drop_last (bool): Whether to drop the last samples for the last batch.
-        shuffle (bool): Whether to shuffle the dataset for each epoch.
     """
-
-    is_train: bool = hp.required("whether to load the training or validation dataset")
-    datadir: str = hp.required("data directory")
-    download: bool = hp.required("whether to download the dataset, if needed")
-    drop_last: bool = hp.optional("Whether to drop the last samples for the last batch", default=True)
-    shuffle: bool = hp.optional("Whether to shuffle the dataset for each epoch", default=True)
+    download: bool = hp.optional("whether to download the dataset, if needed", default=True)
+    synthetic: Optional[SyntheticBatchPairDatasetHparams] = hp.optional(
```

Comment:
Sure, what should it be called?

Reply:
(Would `SyntheticDatasetHparams` work? I suppose that the parameters here should be the same for NLP as they are for images.)

Reply:
I think I still need some convincing that these actually need to be hparams (e.g. have some settings). If the number of settings is small (which it seems to be), would prefer flat:
```
synthetic: bool
synthetic_device: Optional[]
```

rather than create yet another hparams class

Reply:
Sure, I can put them on the synthetic mixin

Reply:
Flattened this and removed the `SyntheticBatchPairDatasetHparams`